File: 3.cpp
#include <iostream>
#include <locale.h>
using namespace std;

// Функция для реверса цифр числа
int reverse_number(int num) {
    int reversed = 0;
    while (num > 0) {
        reversed = reversed * 10 + num % 10; // Добавляем последнюю цифру исходного числа
        num /= 10; // Убираем последнюю цифру
    }
    return reversed;
}

int main() {
    setlocale(LC_ALL, "rus");
    int n;
    // Ввод количества чисел
    cout << "Введите количество чисел (N): ";
    cin >> n;

    // Ввод чисел и обработка
    cout << "Введите " << n << " натуральных чисел через пробел: ";
    for (int i = 0; i < n; ++i) {
        int number;
        cin >> number;

        int reversed = reverse_number(number);
        int difference = number - reversed;

        // Ручной расчет модуля разности
        int absolute = (difference < 0) ? -difference : difference;

        cout << "|" << number << " - " << reversed << "| = "
            << absolute << endl;
    }

    return 0;
}

File: 3.php
<?php

function reverseNumber($num) {
    $reversed = 0;
    while ($num > 0) {
        $reversed = $reversed * 10 + $num % 10;
        $num = (int)($num / 10);
    }
    return $reversed;
}

// Ввод данных
echo "Введите количество чисел: ";
$n = intval(trim(fgets(STDIN)));

echo "Введите числа через пробел: ";
$input = trim(fgets(STDIN));
$numbers = explode(' ', $input);
$numbers = array_slice($numbers, 0, $n); // Обрезаем до нужного количества
echo "Результат: ";
// Обработка и вывод
foreach ($numbers as $numStr) {
    $num = intval($numStr);
    $reversed = reverseNumber($num);
    $difference = abs($num - $reversed);
    echo "|$num - $reversed| = $difference\n";
}

?>
